src/main/groovy/com/avioconsulting/mule/linter/model/ConfigurationFile.groovy

```diff
@@ -23,6 +24,54 @@ class ConfigurationFile extends ProjectFile {
         }
     }
 
+    Boolean doesExists() {
+        return exists
+    }
+
+    List<MuleComponent> findChildComponents(Map<String, String> elements, Boolean include) {
+        List<MuleComponent> componentList = []
+        def childNodes = configXml.childNodes()
+        def comps = []
+        childNodes.each {
+            node ->
+            if (include) {
+                if (checkElementExists( node, elements )) {
+                    comps.add(node)
+                }
+            } else {
+                if (!checkElementExists( node, elements )) {
+                    comps.add(node)
+                }
+            }
+        }
+
+        comps.each { comp ->
+            Map<String, String> atts = [:]
+            comp.attributes.each {
+                atts.put(it.key, it.value)
+            }
+            componentList.add(new MuleComponent(comp.name(), comp.namespaceURI(), atts))
+        }
+        return componentList
+    }
+
+    List<MuleComponent> findGlobalConfigs(Map<String, String> noneGlobalElements) {
+        return findChildComponents(noneGlobalElements, false)
+    }
+
+    Boolean checkElementExists(Node node, Map<String, String> element) {
```

Comment:
Also add groovydocs here, again the purpose of the method isn't clear without it right now.

Reply:
added.